### ISSUE TITLE
[Bug]: Goto failed:Page.goto: Target page, context or browser has been closed\\nCall log:\\n  - navigating to

### ISSUE THREAD

**Issue by Tefile** [Kaliiiiiiiiii-Vinyzu/patchright#217]
### Checklist before reporting

- [x] I have searched for [similar issues](https://github.com/Kaliiiiiiiiii-Vinyzu/patchright/issues?q=sort%3Arelevance%20is%3Aissue) and didn't find a duplicate.
- [x] I have updated to the latest version of Patchright to verify the issue still exists.
- [x] I have verified that this issue does not occur when using [Playwright](https://github.com/microsoft/playwright/) directly.

### Patchright Version

1.60.0

### Operating System

windows

### Bug Description

Connect your Gologin fingerprint browser via CDP, and then use "goto" to open the page.

### Test file (self-contained)

class CustomPage:
    def __init__(self, page: Page, tracking_link, data, platform):
        self.page = page
        self.tracking_link = tracking_link
        self.data = data
        self.is_android = True if platform == "AdsPower" else False
        self._cdp_client = None
        self.redirect_chain = []
        self.request_handler = ""
        self.response_handler = ""
        self.request_fail_handler = ""
        self.BrowserException = BrowserException
        self.PageException = PageException

    @classmethod
    @contextmanager
    def create(cls, ws_url: str, tracking_link: str, resolution: dict, platform: str, data: dict):
        """
        浏览器上下文管理器

        Args:
            ws_url: 指纹浏览器 WebSocke 地址
            tracking_link: 访问链接
            resolution: 屏幕分辨率
            platform: 平台标识（GoLogin / GoLoginPc / AdsPower）
            data: 业务数据

        Yields:
            CustomPage 实例
        """
        sp = sync_playwright().start()
        browser = None
        custom_page = None
        try:
            browser = sp.chromium.connect_over_cdp(
                ws_url,
                timeout=BROWSER_CONNECT_TIMEOUT,
                # slow_mo=100,
            )
            if not browser.contexts:
                raise BrowserException(error_info="get context fail: no contexts")

            page_org = cls._find_usable_page(browser)
            custom_page = cls(page_org, tracking_link, data, platform)
            yield custom_page
        finally:
            # 清理 CDP Session
            if custom_page and custom_page._cdp_client:
                try:
                    custom_page._cdp_client.detach()
                except Exception:
                    pass
            if browser:
                try:
                    browser.close()
                except Exception:
                    pass
            sp.stop()

    @staticmethod
    def _find_usable_page(browser) -> Page:
        """从浏览器上下文中找到可用的页面"""
        last_error = None
        for context in browser.contexts:
            try:
                pages = context.pages
                page = pages[0] if pages else context.new_page()
                if page.is_closed():
                    page = context.new_page()
                # 探测页面是否可用
                page.wait_for_timeout(100)
                page.evaluate("() => document.readyState")
                return page
            except Exception as e:
                last_error = e
        raise BrowserException(error_info=f"no valid context/page found: {last_error}")

    @property
    def cdp_mobile_session(self):
        if self._cdp_client is None and self.is_android:
            self._cdp_client = self.page.context.new_cdp_session(self.page)
            self._cdp_client.send("Emulation.setTouchEmulationEnabled", {
                "enabled": True,
                "configuration": "mobile"
            })
        return self._cdp_client

    def network_control(self):
        """
        确保重定向稳定 确保无导航请求 必须在goto后
        :return:
        req.is_navigation_request 只监测导航不监测其他资源
        """
        for _ in range(5):
            try:
                self.page.wait_for_event("request", predicate=lambda req: req.is_navigation_request(), timeout=5_000)
            except Exception as e:
                break
        for _ in range(5):
            try:
                self.page.wait_for_event("response", predicate=lambda res: res.request.resource_type == "document",
                                         timeout=5_000)
            except Exception as e:
                break

    def ex_redirect_chain(self):
        """
        检查重定向链 必须在goto后
        :return:
        """
        final_url = self.page.url
        for i, entry in enumerate(self.redirect_chain):
            status = entry.get("status", "unknown")
            # 检测当前失败链接是不是最终导航栏url 是则直接退出
            if status == "failed" and final_url == entry['url']:
                result = {
                    "error": f"Redirect Fail: {entry['url']} - {entry.get('error', 'unknown')}",
                    "failed_step": i,
                    "failed_url": entry["url"]
                }
                sys.exit(json.dumps(result))

            if isinstance(status, int) and status >= 400:
                result = {
                    "error": f"HTTP Fail {status} at {entry['url']}",
                    "failed_step": i,
                    "failed_url": entry["url"]
                }
                sys.exit(json.dumps(result))

    def check_page_body(self) -> None:
        """
        检查最终页面内容是否正常
        """
        self.page.wait_for_load_state("domcontentloaded", timeout=PAGE_LOAD_TIMEOUT)

        body_el = self.page.query_selector("body")
        if not body_el:
            raise PageException(error_info="Body element not found", error_url=self.tracking_link)
        content = self.page.query_selector('body').text_content()
        error_info = ""
        if self.page.url == "" or self.page.url == "about:blank":
            error_info = 'The link is empty or it is "about:blank".'
        elif not content or not content.strip():
            error_info = 'The final page content is empty'
        elif "ERR_SSL_PROTOCOL_ERROR" in content:
            error_info = "The final page content is ERR_SSL_PROTOCOL_ERROR"
        elif "ERR_CONNECTION_" in content:
            error_info = f"The final page content is ERR_CONNECTION:{content}"
        if error_info:
            raise PageException(error_info=error_info, error_url=self.tracking_link)
        # 检查是否为纯 JSON 响应
        try:
            json.loads(content)
            raise HtmlException(error_info=f"html is json: content={content}", error_url=self.tracking_link)
        except (json.JSONDecodeError, ValueError):
            pass

    def custom_wait(self):
        # 默认跳转等待
        self.page.wait_for_timeout(1000)
        self.page.wait_for_load_state('domcontentloaded')
    def goto(self):
        """
        打开浏览器
        :return:
        """
        try:
            self.page.goto(self.tracking_link, wait_until='domcontentloaded', timeout=PAGE_LOAD_TIMEOUT)
        except Exception as e:
            raise PageException(error_info=f"Goto failed:{e}", error_url=self.tracking_link)

**Comment by Vinyzu** [comment]
Patchright doesnt support any external third party paid antidetect browsers like Gologin. They often tweak the CDP which makes them unreliable if not impossible to work with.
Please open a new issue if this error persists with a free "real" browser, e.g. Google Chrome Stable.

Also see https://stackoverflow.com/help/minimal-reproducible-example
